### PR TITLE
feat: Add style transform support for stencils

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Fixes: #2463 

* Refactors the style transform to make it easier to understand the parts.
* Add automated fallback support
* Removed unnecessary fallbacks
* Add style transform support for `createStencils`
* Add auto-wrapping for CSS variables in `createStyles`, `createStencils`, and the transformer.

## Release Category
Components

### Release Note

CSS variables can now be unwrapped anywhere in a property definition when using `createStyles` or `createStencil`. Previously, unwrapped variables could only be used if they started the property. For example, the following is now valid:

```ts
const myVar = '--my-var';

const myStyles= createStyles({
  boxShadow: `0 0 0 2px ${myVar}`
})
```

You no longer need to use `cssVar` around `myVar` to be valid. This should help with the transition from hard-coded variables (`const myVar = 'red' => const myVar = '--my-var'`) without needing to wrap in a `cssVar`. It allows us to think about `myVar` as a JS variable and the system will convert to a CSS variable for us.

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Areas for Feedback? (optional)

- [x] Code
- [x] Documentation
- [x] Testing

## Testing Manually

Tests should suffice, but you could always do a `(cd modules/react; yarn clean && yarn build:es6)` and verify the `dist` directory. Remember to do a `yarn clean` when you're done otherwise your environment is hosed.

### CODE REVIEW

File: modules/preview-react/form-field/lib/FormFieldLabel.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const asteriskStyles = createStyles({
   fontSize: type.properties.fontSizes[20],
   fontWeight: type.properties.fontWeights.regular,
   textDecoration: 'unset',
-  color: cssVar(brand.error.base, '#de2e21'),
+  color: brand.error.base,
```
Comment (NicholasBoll) [inline modules/preview-react/form-field/lib/FormFieldLabel.tsx:28]: This type of change is in many places. Auto-fallback works and manually including a fallback is a maitenance issue.

**Review by mannycarrera4 — APPROVED** [review]
You write code, sometimes I agree with the code you write

File: modules/styling-transform/lib/utils/handleFocusRing.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            defaults[property.name.text as keyof typeof defaults] = property.initializer;
+          }
+        });
+      }
+
+      const inset = parseNodeToStaticValue(defaults.inset, checker, prefix, vars);
+
+      let boxShadow: ts.TemplateExpression;
+      switch (inset) {
+        case 'outer':
```
Comment (josh-bagwell) [inline modules/styling-transform/lib/utils/handleFocusRing.ts:53]: Would `outer` be considered to be both focus rings on the "outside" of the container? I noticed an `inset` here or this something we need to talk about when re-factoring `focusRing`?

Comment (NicholasBoll) [reply modules/styling-transform/lib/utils/handleFocusRing.ts:53]: This is why I think `focusRing` will need an update. `outer` is actually both inner:

https://github.com/Workday/canvas-kit/blob/8817a0a5d258a0f1302ed563beeaa70b40883e5e/modules/react/common/lib/styles/focusRing.ts#L10-L13

I'm supporting `focusRing` in the static optimizer, not changing it.

**Pull request merged**